### ISSUE TITLE
Output of the  train_test_split_edges function in the torch_geometric.utils package

### ISSUE THREAD

**Issue by gamzeakyol** [rusty1s/pytorch_geometric#1806]
## ❓ Questions & Help

Hi, 

In torch_geometric.utils package, output of the function train_test_split_edges(data, val_ratio=0.05, test_ratio=0.1) is the following:

Data(edge_attr=[200, 3], test_neg_edge_index=[2, 14], test_pos_edge_index=[2, 14], train_neg_adj_mask=[39, 39], train_pos_edge_index=[2, 28], val_neg_edge_index=[2, 1], val_pos_edge_index=[2, 1], x=[39, 21]).

I cannot obtain train_neg_edge_index value from this output. How can I obtain it? Is there a bug about it?

Thank you.

**Comment by rusty1s** [comment]
In generally, one can use any edge not in `train_pos_edge_index` as negative edges during training. These edges are marked in `train_neg_adj_mask`. As an alternative, you can use `torch_geometric.utils.negative_sampling` to sample a subset of negative training edges.

**Comment by gamzeakyol** [comment]
Thank you for your quick response! I will try this solution.

**Comment by Qbuer** [comment]
> In generally, one can use any edge not in `train_pos_edge_index` as negative edges during training. These edges are marked in `train_neg_adj_mask`. As an alternative, you can use `torch_geometric.utils.negative_sampling` to sample a subset of negative training edges.

Will the positive examples in the test set be treated as negative examples during negative sampling?

**Comment by rusty1s** [comment]
Yeah, that is possible, but not avoidable. You cannot exclude them as negative samples since technically you are not aware of their existence.

**Comment by Qbuer** [comment]
Yeah that sounds reasonable. Thanks for the repley.

**Comment by gamzeakyol** [comment]
In train_neg_adj_mask matrix, is True represents the negative edges or the positive edges?
Thanks.
